### ISSUE TITLE
Dependencies not properly bundled

### ISSUE THREAD

**Issue by BlueGlassBlock** [YunYouJun/valaxy#479]
### Describe the bug

Dependencies are not properly bundled by vite, causing a 25MB load & ~1200 requests on a typical theme-yun blog.

<details>
<summary><code>pnpm dev</code> log</summary>

```
> valaxy-reprod@0.0.0 dev D:\Projects\valaxy-reprod
> valaxy

◐ Resolve valaxy config ...                                                                                                                                   18:00:19
✔ Resolve userValaxyConfig from D:/Projects/valaxy-reprod/valaxy.config.ts 167.50ms                                                                          18:00:20
✔ Resolve siteConfig from D:/Projects/valaxy-reprod/site.config.ts 168.22ms                                                                                  18:00:20
✔ Resolve valaxy.config.ts from theme(yun) 392.27ms                                                                                                          18:00:20
✔ Resolve addons from D:/Projects/valaxy-reprod
  vite:deps removing old cache dir D:/Projects/valaxy-reprod/node_modules/.pnpm/valaxy@0.20.4_@babel+parser@7.26.2_@nuxt+kit@3.14.1592_rollup@4.27.4__@types+markdown-it@14.1_kafwzngjmitkdxfuy77m2q7wlm/node_modules/valaxy/node_modules/.vite/deps +0ms
  vite:deps scanning for dependencies... +0ms
✔ valaxy: server ready                                                                                                                                       18:00:26

  🌌 Valaxy  v0.20.4

  🪐 theme   > yun (v0.20.4)
  📁 D:\Projects\valaxy-reprod

  Preview    > http://localhost:4860/
  Network    > http://198.18.0.1:4860/
  Network    > http://100.72.120.18:4860/
  Network    > http://10.129.195.177:4860/
  Network    > http://172.30.48.1:4860/
  Network    > http://192.168.176.1:4860/

  shortcuts  > restart | open | qr | edit

  vite:deps Scan completed in 130.24ms: no dependencies found +0ms
  vite:deps creating package.json in D:/Projects/valaxy-reprod/node_modules/.pnpm/valaxy@0.20.4_@babel+parser@7.26.2_@nuxt+kit@3.14.1592_rollup@4.27.4__@types+markdown-it@14.1_kafwzngjmitkdxfuy77m2q7wlm/node_modules/valaxy/node_modules/.vite/deps_temp_fc5d2617 +361ms
  vite:deps Dependencies bundled in 2323.62ms +2s
  vite:deps ✨ static imports crawl ended +42s
  vite:deps ✨ using post-scan optimizer result, the scanner found every used dependency +1ms
  vite:deps creating _metadata.json in D:/Projects/valaxy-reprod/node_modules/.pnpm/valaxy@0.20.4_@babel+parser@7.26.2_@nuxt+kit@3.14.1592_rollup@4.27.4__@types+markdown-it@14.1_kafwzngjmitkdxfuy77m2q7wlm/node_modules/valaxy/node_modules/.vite/deps_temp_fc5d2617 +40s
  vite:deps renaming D:/Projects/valaxy-reprod/node_modules/.pnpm/valaxy@0.20.4_@babel+parser@7.26.2_@nuxt+kit@3.14.1592_rollup@4.27.4__@types+markdown-it@14.1_kafwzngjmitkdxfuy77m2q7wlm/node_modules/valaxy/node_modules/.vite/deps_temp_fc5d2617 to D:/Projects/valaxy-reprod/node_modules/.pnpm/valaxy@0.20.4_@babel+parser@7.26.2_@nuxt+kit@3.14.1592_rollup@4.27.4__@types+markdown-it@14.1_kafwzngjmitkdxfuy77m2q7wlm/node_modules/valaxy/node_modules/.vite/deps +3ms
  vite:deps ✨ dependencies optimized +5ms
```

</details>

<details>
<summary>Vite Metadata</summary>

```json
{
  "hash": "4ad7befc",
  "configHash": "082786c5",
  "lockfileHash": "66419126",
  "browserHash": "6b63d93f",
  "optimized": {
    "@ctrl/tinycolor": {
      "src": "../../../../../../@ctrl+tinycolor@4.1.0/node_modules/@ctrl/tinycolor/dist/module/public_api.js",
      "file": "@ctrl_tinycolor.js",
      "fileHash": "138f0c5f",
      "needsInterop": false
    },
    "gsap/dist/ScrollToPlugin": {
      "src": "../../../../../../gsap@3.12.5/node_modules/gsap/dist/ScrollToPlugin.js",
      "file": "gsap_dist_ScrollToPlugin.js",
      "fileHash": "ed16af50",
      "needsInterop": true
    },
    "@unhead/schema-org": {
      "src": "../../../../../../@unhead+schema-org@1.11.13_@unhead+vue@1.11.13_vue@3.5.13_typescript@5.7.2___unhead@1.11.13/node_modules/@unhead/schema-org/dist/index.mjs",
      "file": "@unhead_schema-org.js",
      "fileHash": "54c7bd8e",
      "needsInterop": false
    },
    "@unhead/vue": {
      "src": "../../../../../../@unhead+vue@1.11.13_vue@3.5.13_typescript@5.7.2_/node_modules/@unhead/vue/dist/index.mjs",
      "file": "@unhead_vue.js",
      "fileHash": "bfabb99a",
      "needsInterop": false
    },
    "@vueuse/integrations/useFuse": {
      "src": "../../../../../../@vueuse+integrations@11.3.0_axios@1.7.8_fuse.js@7.0.0_nprogress@0.2.0_qrcode@1.5.4_vue@3.5.13_typescript@5.7.2_/node_modules/@vueuse/integrations/useFuse.mjs",
      "file": "@vueuse_integrations_useFuse.js",
      "fileHash": "711f2b3e",
      "needsInterop": false
    },
    "defu": {
      "src": "../../../../../../defu@6.1.4/node_modules/defu/dist/defu.mjs",
      "file": "defu.js",
      "fileHash": "0e4d3f9c",
      "needsInterop": false
    },
    "katex": {
      "src": "../../../../../../katex@0.16.11/node_modules/katex/dist/katex.mjs",
      "file": "katex.js",
      "fileHash": "31f94001",
      "needsInterop": false
    },
    "nprogress": {
      "src": "../../../../../../nprogress@0.2.0/node_modules/nprogress/nprogress.js",
      "file": "nprogress.js",
      "fileHash": "b6c949c9",
      "needsInterop": true
    },
    "vue-i18n": {
      "src": "../../../../../../vue-i18n@10.0.4_vue@3.5.13_typescript@5.7.2_/node_modules/vue-i18n/dist/vue-i18n.mjs",
      "file": "vue-i18n.js",
      "fileHash": "1e362723",
      "needsInterop": false
    },
    "fuse.js": {
      "src": "../../../../../../fuse.js@7.0.0/node_modules/fuse.js/dist/fuse.mjs",
      "file": "fuse__js.js",
      "fileHash": "9e0b3851",
      "needsInterop": false
    },
    "medium-zoom": {
      "src": "../../../../../../medium-zoom@1.1.0/node_modules/medium-zoom/dist/medium-zoom.esm.js",
      "file": "medium-zoom.js",
      "fileHash": "6fa1224b",
      "needsInterop": false
    },
    "vanilla-lazyload": {
      "src": "../../../../../../vanilla-lazyload@19.1.3/node_modules/vanilla-lazyload/dist/esm/lazyload.js",
      "file": "vanilla-lazyload.js",
      "fileHash": "a9127c46",
      "needsInterop": false
    }
  },
  "chunks": {
    "chunk-4WGWJTZB": {
      "file": "chunk-4WGWJTZB.js"
    },
    "chunk-J62M3TMS": {
      "file": "chunk-J62M3TMS.js"
    },
    "chunk-BUSYA2B4": {
      "file": "chunk-BUSYA2B4.js"
    }
  }
}
```

</details>

### Reproduction

pnpm create valaxy

### System Info

```Shell
System:
    OS: Windows 11 10.0.26100
    CPU: (16) x64 AMD Ryzen 7 5800H with Radeon Graphics
    Memory: 30.70 GB / 63.86 GB
  Binaries:
    Node: 22.3.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.19.2 - C:\Program Files\nodejs\npm.CMD
    pnpm: 9.7.0 - C:\Program Files\nodejs\pnpm.CMD
  Browsers:
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.26100.1882
```

### Used Package Manager

pnpm

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by BlueGlassBlock** [comment]
Also confirmed to be reproducible on a standard codespace of [YunYouJun/yunyoujun.github.io](https://github.com/YunYouJun/yunyoujun.github.io)

**Comment by BlueGlassBlock** [comment]
Updating optimizeDeps, or simply reducing full-size import should solve this issue.

**Comment by YunYouJun** [comment]
https://vite.dev/guide/dep-pre-bundling.html#the-why

`date-fns` is special in that it packages the file into multiple subfiles.
This makes the shortcomings even more magnified on `Windows`.

```ts
import { setDefaultOptions } from 'date-fns/setDefaultOptions'
import { zhCN } from 'date-fns/locale/zh-CN'
import { enUS } from 'date-fns/locale/en-US'
```

**Comment by YunYouJun** [comment]
Fixed in `0.20.6`.
